File: app/src/main/java/br/com/inngage/sdk/InngageWebViewActivity.java
package br.com.inngage.sdk;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.webkit.WebView;
import android.webkit.WebViewClient;

public class InngageWebViewActivity extends AppCompatActivity {
    private class MyWebViewClient extends WebViewClient {
        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            view.loadUrl(url);
            return true;
        }
    }
    private static final String TAG = InngageConstants.TAG;
    private String name;
    private WebView webview;
    String url ="";
    @SuppressLint("SetJavaScriptEnabled")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);




        }
    public void web(String url) {


        // Bundle bundle = getIntent().getExtras();

//        if (getIntent().hasExtra("EXTRA_URL")) {
//
//            url = bundle.getString("EXTRA_URL");
//        }


        Log.d(TAG, "Opening WebView component");


        Log.d(TAG, "Opening URL: " + url);
        webview = findViewById(R.id.inn_webview);
        webview.setWebViewClient(new MyWebViewClient());
        webview.loadUrl(url);
        webview.requestFocus();
    }


    @Override
    protected void onDestroy(){
        super.onDestroy();

        if (webview != null) {
            webview.destroy();
        }
    }
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.web_view);
//
//        Log.d(TAG, "Calling InngageWebViewActivity...");
//
//        //Intent intent = getIntent();
//        //Bundle bundle  = intent.getExtras();
//
//
//        //Log.d(TAG, "Calling InngageWebViewActivity to open URL: " + bundle.getString("URL"));
//
//
//        String url = null;
//
//        Intent intent = getIntent();
//        Bundle bundle  = intent.getExtras();
//
//        if (bundle.getString("url") != null) {
//
//            Log.d(TAG, "Calling InngageWebViewActivity to open URL: " + bundle.getString("url"));
//        }
//        if (getIntent().hasExtra("URL")) {
//          url = getIntent().getStringExtra("URL");
//            Log.d(TAG, "Calling InngageWebViewActivity to open URL: " + name);
//        } else {
//            Log.d(TAG, "Activity cannot find  extras ");
//        }
//
//        //String url = "";
//
//        //Intent intent = getIntent();
//        //Bundle bundle  = intent.getExtras();
//
//        Log.d(TAG, "Opening WebView component");
//
//        if ((bundle != null) && (bundle.getString("URL") != null)) {
//
//            url = bundle.getString("URL");
//
//            Log.d(TAG, "Opening URL: " + url);
//
//            WebView myWebView = findViewById(R.id.webview);
//
//            if (myWebView != null) {
//
//                myWebView.loadUrl(url);
//
//            } else {
//
//                Log.d(TAG, "WebView object no has a instance");
//            }
//        }
//
//    }
}

File: src/main/kotlin/br/com/inngage/sdk/internal/service/inapp/InngageWebViewActivity.kt
package br.com.inngage.sdk.internal.service.inapp

import android.os.Bundle
import android.util.Log
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import br.com.inngage.sdk.R
import br.com.inngage.sdk.internal.core.config.InngageConfig

/**
 * Displays a URL inside a WebView.
 * Replaces InngageWebViewActivity.java — pure Kotlin, no behaviour change.
 */
internal class InngageWebViewActivity : AppCompatActivity() {

    private val tag = InngageConfig.TAG_NOTIFY
    private var webView: WebView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.web_view)
        webView = findViewById(R.id.inn_webview)

        intent.getStringExtra(EXTRA_URL)?.let { loadUrl(it) }
    }

    private fun loadUrl(url: String) {
        Log.d(tag, "Opening WebView URL: $url")
        webView?.apply {
            webViewClient = object : WebViewClient() {
                override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                    view.loadUrl(url)
                    return false
                }
            }
            loadUrl(url)
            requestFocus()
        }
    }

    override fun onDestroy() {
        webView?.destroy()
        webView = null
        super.onDestroy()
    }

    companion object {
        const val EXTRA_URL = "extra_url"
    }
}

